arch/x86/x86_intrins.h

```diff
@@ -89,4 +89,40 @@ static inline __m512i _mm512_zextsi128_si512(__m128i a) {
 #  undef _mm512_extracti32x4_epi32
 #  define _mm512_extracti32x4_epi32(v1, e1) _mm512_maskz_extracti32x4_epi32(UINT8_MAX, v1, e1)
 #endif
+
+#if defined(_MSC_VER) && !defined(__clang__)
+#include <intrin.h>
+/* For whatever reason this intrinsic is 64 bit only with MSVC?
+ * While we don't have 64 bit GPRs, it should at least be able to move it to stack
+ * or shuffle it over 2 registers */
+#if !defined(_M_AMD64)
+/* So, while we can't move directly to a GPR, hopefully this move to
+ * a stack resident variable doesn't equate to something awful */
+static inline int64_t _mm_cvtsi128_si64x(__m128i a) {
+    union { __m128i v; int64_t i; } u;
+    u.v = a;
+    return u.i;
+}
+
+static inline __m128i _mm_cvtsi64x_si128(int64_t a) {
```

Comment:
v142 toolchain can't handle inlining this function... It works without `inline` but compiles to really slow code. I tried converting it to preprocessor macro, but that didn't work either.

Reply:
Any word yet if and when Microsoft will actually fix this?

Reply:
Not yet... They're just guessing how to write code to workaround the bug instead of actually trying to fix the issue. It might take a while for them to reply due to weekend and time zones...

They don't really read the zlib-ng code further than the function where the issue happens and thus suggest quite pointless workarounds. I had to restrict audience of the latest comment I wrote on the support forum, because it was not nice thing to say... But it was the only way to nudge them to actually try to fix the issue and not suggest workaround I have already tried and deemed unsuitable.